### DISCUSSION TITLE
Are there any pre-steps before installing docker on a new Proxmox Installation?

### DISCUSSION THREAD

**Discussion by siniradam** [tteck/Proxmox#2583] [Q&A] [answered]
I'm pretty new to proxmox, used a video tutorial on youtube. Succesfuly installed, ubuntu, windows, TrueNAS.
When I try to install Docker w/ this 

```bash
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/docker.sh)"
```

Wizard starts, and pick values like this;
- var_disk="20"
- var_cpu="3"
- var_ram="3072"
- var_os="ubuntu"
- var_version="23.04"

After the last step, I'm getting this error message;
`"Unable to detect a valid Container Storage location"`

Tried running 
```bash
pvesm status -content rootdir | awk 'NR>1'
```
I doesn't return anything, when I try the next command, the result is below;
```bash
pvesm status -content vztmpl | awk 'NR>1'
local         dir     active       951541116        52943588       858882592    5.56%
```

This command `cat /sys/module/kvm_amd/parameters/nested` returns 1 which should mean virtualization is enabled.

```bash
proxmox-ve: 8.1.0 (running kernel: 6.5.13-1-pve)
pve-manager: 8.1.4 (running version: 8.1.4/ec5affc9e41f1d79)
```

System Details goes like this;
- BIOS Model name:     AMD Ryzen 7 5800H with Radeon Graphics          Unknown CPU @ 3.2GHz
- Virtualization:        AMD-V

Am I missing a step, are there any other steps that I need to take before creating Docker Container?

**Comment by tteck** [comment]
By any chance, was that video by NetworkChuck?

**Reply by siniradam** [reply]
Indeed it is NetworkChuck.

**Reply by tteck** [reply]
Based on my advice, **reinstalling Proxmox is necessary**. It seems that following NetworkChuck's suggestion to delete the local-lvm storage has led you astray, as you're now discovering.

**Reply by siniradam** [reply]
I knew something was wrong. It didn't feel right. Would you care to explain what's going on behind the scenes briefly or can you point me in the direction of what I need to search for?

Aside from reinstalling, if there is a way to fix it by creating local-lvm again, would that be a bad idea? Considering I've already installed other VMs.

**Reply by tteck** [reply]
Local-LVM storage typically provides better performance compared to standard file-based storage, as it uses block devices directly. This can lead to faster read and write operations, which is especially important for I/O-intensive workloads.

With Local-LVM, you can isolate storage for each virtual machine or container using logical volumes. This helps prevent one VM or container from impacting others in terms of storage I/O or running out of disk space.

Using Local-LVM helps reduce file system fragmentation, which can improve performance and storage efficiency.

**Comment by tteck** [comment]
What does the command `pvesm status` show?

**Reply by siniradam** [reply]
The result is as below.
```shell
Name         Type     Status           Total            Used       Available        %
local         dir     active       951541116        52966760       858859420    5.57%
```

**Reply by tteck** [reply, marked as answer]
Never delete local-lvm!
